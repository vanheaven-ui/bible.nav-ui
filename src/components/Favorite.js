import PropTypes from 'prop-types';
import '../styles/Favorite.css';

const Favorite = ({ favorites, removeFavorite }) => (
  <>
    { favorites.length > 0 ? (favorites.map(favorite => (
      <article key={favorite.id}>
        <div className="heading">
          <h2>
            {favorite.book_name}
            <span>{favorite.chapter_num}</span>
            <span>{favorite.verse_num}</span>
          </h2>
          <button type="button" onClick={() => removeFavorite(favorite.id)}>
            <i className="fas fa-trash" />
          </button>
        </div>
        <p>{favorite.verse}</p>
      </article>
    ))) : <p id="no__favorites">No favorites yet</p>}
  </>
);

Favorite.propTypes = {
  favorites: PropTypes.arrayOf(
    PropTypes.shape({
      book_name: PropTypes.string.isRequired,
      chapter_num: PropTypes.string.isRequired,
      verse_num: PropTypes.string.isRequired,
      verse: PropTypes.string.isRequired,
    }),
  ).isRequired,
  removeFavorite: PropTypes.func.isRequired,
};

export default Favorite;
