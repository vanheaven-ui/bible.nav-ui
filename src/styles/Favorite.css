article {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.heading {
  width: 80%;
  display: flex;
  justify-content: space-between;
  padding: 0 10px;
  margin-top: 20px;
}

.heading h2 {
  align-self: flex-start;
}

.title {
  font-size: 28px;
  text-shadow: 1px 1px 2px #fff;
}

.title span {
  color: #fff;
  font-size: 24px;
  text-shadow: 1px 1px 2px #111;
}

.title span:first-child {
  margin-left: 10px;
}

.heading button {
  border: none;
  background-color: transparent;
  transition: all 0.5s ease-in-out;
}

article p {
  background-color: #fff;
  padding: 20px 30px 20px 10px;
  width: 80%;
  border-radius: 10px;
  color: #666;
  font-weight: 500;
}

.fa-trash {
  color: red;
  text-shadow: 1px 1px 2px #111;
  font-size: 20px;
  transition: all 0.5s ease;
}

.fa-trash:hover {
  color: #fff;
}

#no__favorites {
  width: 80%;
  display: flex;
  justify-content: center;
  margin: 0 auto;
}
