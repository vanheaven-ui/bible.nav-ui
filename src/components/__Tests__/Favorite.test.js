import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import rootDiv from '../../testHelper';
import '@testing-library/jest-dom';
import Favorite from '../Favorite';

describe('Favorite Component test', () => {
  const favorites = [];
  const removeFavorite = () => '';

  afterEach(cleanup);

  it('renders without crashing', () => {
    ReactDOM.render(
      <Favorite favorites={favorites} removeFavorite={removeFavorite} />, rootDiv(),
    );
  });

  it('does not change unexpectedly', () => {
    const tree = renderer.create(
      <Favorite favorites={favorites} removeFavorite={removeFavorite} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
