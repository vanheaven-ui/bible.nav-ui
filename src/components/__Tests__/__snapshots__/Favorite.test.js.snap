// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Component test does not change unexpectedly 1`] = `
<p
  id="no__favorites"
>
  No favorites yet
</p>
`;
